tools/crm_mon.c

```diff
@@ -1033,8 +1125,13 @@ detect_user_input(GIOChannel *channel, GIOCondition condition, gpointer user_dat
     }
 
 refresh:
-    fencing_connect();
-    rc = cib_connect(FALSE);
+    rc = pacemakerd_status();
+    if (rc == pcmk_rc_ok) {
+        fencing_connect();
+        rc = cib_connect(FALSE);
+    } else {
+        out->err(out, "CIB and fencer sub-daemons not running");
```

Comment:
I wouldn't consider this an error, it's more like the "Waiting until cluster is available" message

Reply:
True but till now we had an output on the error-level complaining that it couldn't connect to CIB and the idea was to still issue an error thus.

Reply:
Hmm, I think we want to show an error if a connection actually fails, but just a message if the daemon appears to not be running. In other words it's not an error for someone to stop or restart their cluster, but it's an error if we run into permission problems, system resource issues, etc.

The goal would be that if we are in a known state, we show that to the user. Of course if a daemon isn't accepting connections we don't necessarily know the user intentionally stopped it, but we can treat it as informational rather than an error.

Reply:
Ok getting rid of that entirely. And possibly move details to handle_connection_failures ... see below.